src/web/controllers/user/event_user_controller.py

```diff
@@ -206,23 +206,23 @@ def _user_event_refresh_needed(
             if screen.last_update > date:
                 return True
             for screen_set in screen.screen_sets_by_id.values():
-                if screen_set.last_update and screen_set.last_update > date:
-                    return True
-                if screen_set.tournament.last_update > date:
+                tournament = screen_set.tournament
+                if (
+                    max(
+                        screen_set.last_update or 0,
+                        tournament.last_update,
+                        tournament.last_player_update,
+                    )
+                    > date
+                ):
                     return True
                 if screen.type in [
                     ScreenType.BOARDS,
                     ScreenType.INPUT,
                     ScreenType.RANKING,
                 ]:
-                    if screen_set.tournament.last_illegal_move_update > date:
-                        return True
-                    if screen_set.tournament.last_result_update > date:
+                    if tournament.last_pairing_update > date:
                         return True
-                    if screen_set.tournament.last_update > date:
```

Comment:
I think you need to check for updates to the tournament to react to screen config changes (new screens, timers, menu changes...)